### ISSUE TITLE
JaCoCo Agent Override

### ISSUE THREAD

**Issue by wakawaka54** [Kotlin/kotlinx-kover#195]
I'm working through JaCoCo support with this tool and I did have quite a bit of trouble with the JaCoCo Agent resolution logic. In the internal builds in my company we provide the JaCoCo dependencies in a custom manner.

This logic here expects the dependency name to be "org.jacoco.agent" https://github.com/Kotlin/kotlinx-kover/blob/main/src/main/kotlin/kotlinx/kover/engines/jacoco/JacocoAgent.kt#L20 . The problem I couldn't find a good way to manually provide an agent jar that would work resolve properly here.

I had to do this:

```
class JacocoAgentDependency(collection: FileCollectionInternal) : DefaultSelfResolvingDependency(collection) {
    override fun getName(): String = "org.jacoco.agent"
}

val koverJacocoConfig = project.configurations.getByName("JacocoKoverConfig")

val jacocoAgentJar: File = // my custom agent jar //

koverJacocoConfig.withDependencies {
    val fileCollectionFactory: FileCollectionFactory = serviceOf()
    add(JacocoAgentDependency(fileCollectionFactory.fixed(jacocoAgentJar)))
}
```

**Comment by shanshin** [comment]
Hi, 
may you clarify, you are using a fork of the JaCoCo?

**Comment by wakawaka54** [comment]
Uhhh technically no. But I think a good way to think about this request is that if I have the JaCoCo Agent JAR located locally within the same project, I'd want to configure Kover to just use that JaCoCo Agent rather than trying to implicitly resolve it.

This might also be better served by allowing some customization into the Kover engine adapters, potentially allowing me to register my own engine, etc.

**Comment by shanshin** [comment]
Resolved in `0.7.0-Alpha`
